In [2]:
import torch

# 自动求导

In [3]:
x = torch.ones((2,2),requires_grad = True)
x

tensor([[1., 1.],
        [1., 1.]], requires_grad=True)

In [6]:
y = x + 2
z = y * y * 3
out = z.mean()
out

tensor(27., grad_fn=<MeanBackward0>)

In [8]:
# 对out计算梯度
out.backward()
x.grad

tensor([[4.5000, 4.5000],
        [4.5000, 4.5000]])

$out = \frac{1}{4}\sum_i 3(x_i+2)^2$  

$\frac{\partial out}{\partial x_i} = \frac{3}{2}(x_i+2)$  

$x_i = 1$  

$\frac{\partial out}{\partial x_i} = 4.5$